src/stores/room-list/filters/SpaceFilterCondition.ts

```diff
@@ -42,10 +42,16 @@ export class SpaceFilterCondition extends EventEmitter implements IFilterConditi
 
     private onStoreUpdate = async (): Promise<void> => {
         const beforeRoomIds = this.roomIds;
-        this.roomIds = SpaceStore.instance.getSpaceFilteredRoomIds(this.space);
+        // clone the set as it may be mutated by the space store internally
+        this.roomIds = new Set(SpaceStore.instance.getSpaceFilteredRoomIds(this.space));
 
         if (setHasDiff(beforeRoomIds, this.roomIds)) {
             this.emit(FILTER_CHANGED);
+            // XXX: Room List Store has a bug where updates to the pre-filter during a local echo of a
+            // tags transition seem to be ignored, so refire in the next tick to work around it
+            setImmediate(() => {
+                this.emit(FILTER_CHANGED);
```

Comment:
I'm not too sure to understand the reason why we have to emit the `FILTER_CHANGED` event twice?
Could you give me a bit of extra context here?

Reply:
The comment above was hoping to cover it,

Without it the room list will ignore this prefilter update and use its internal local echo instead, causing the room to stay in Rooms until the next thing triggers the roomlist to recalculate.

This workaround already existed prior to #5825, that PR removed it assuming that it resolved all such cases, but it did not